### PR TITLE
istioctl: add lightstep analyzer

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fixes: https://github.com/istio/istio/issues/40027

### CODE REVIEW

File: pkg/config/analysis/msg/messages.yaml [hunk 1]
```diff
@@ -644,3 +644,13 @@ messages:
         type: "[]string"
       - name: namespace
         type: string
+
+  - name: "LightstepProviderStillUsed"
```
Comment (hanxiaop) [inline pkg/config/analysis/msg/messages.yaml:648]: Maybe `DeprecatedLightstepProvider` is better?

File: pkg/config/analysis/msg/messages.yaml [hunk 2]
```diff
@@ -644,3 +644,13 @@ messages:
...
+    code: IST0161
+    level: Warning
+    description: "Lightstep provider still used"
```
Comment (hanxiaop) [inline pkg/config/analysis/msg/messages.yaml:651]:
```suggestion
    description: "Lightstep provider is still being used"
```

File: releasenotes/notes/43998.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+apiVersion: release-notes/v2
+kind: feature
+area: telemetry
```
Comment (hanxiaop) [inline releasenotes/notes/43998.yaml:4]: This should be `istioctl`

File: releasenotes/notes/43998.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+issue:
+  - 40027
+
+releaseNotes:
+- |
+  **Added** analyzer for lightstep provider.
```
Comment (hanxiaop) [inline releasenotes/notes/43998.yaml:10]: Can you add more details for the analyzer?

File: pkg/config/analysis/msg/messages.yaml [hunk 5]
```diff
@@ -423,7 +423,7 @@ messages:
       - name: ports
         type: "[]int32"
 
-# https://github.com/envoyproxy/envoy/issues/6767
+  # https://github.com/envoyproxy/envoy/issues/6767
```
Comment (hanxiaop) [inline pkg/config/analysis/msg/messages.yaml:426]: Keep the same format as before?

File: pkg/config/analysis/msg/messages.yaml [hunk 6]
```diff
@@ -644,3 +644,13 @@ messages:
         type: "[]string"
       - name: namespace
         type: string
+
+  - name: "DeprecatedLightstepProvider"
+    code: IST0161
+    level: Warning
+    description: "Lightstep provider is still beging used"
```
Comment (hanxiaop) [inline pkg/config/analysis/msg/messages.yaml:651]:
```suggestion
    description: "Lightstep provider is still being used"
```

**Review by hanxiaop — COMMENTED** [review]
LGTM after resolving the comments

**Review by GregHanson — APPROVED** [review]

File: pkg/config/analysis/msg/messages.yaml [hunk 7]
```diff
@@ -644,3 +644,13 @@ messages:
         type: "[]string"
       - name: namespace
         type: string
+
+  - name: "DeprecatedLightstepProvider"
```
Comment (howardjohn) [inline pkg/config/analysis/msg/messages.yaml:648]: We already have a general "Deprecated" message, do we need another?

Comment (zirain) [reply pkg/config/analysis/msg/messages.yaml:648]: I will revise later.

**Pull request merged**